java/arcs/sdk/Particle.kt

```diff
@@ -25,6 +26,11 @@ typealias HandleHolder = arcs.core.host.api.HandleHolder
 /** Base interface for all particles. */
 typealias Particle = Particle
 
+/** A convenience method for moving onto the dispatcher associated with a particle. */
+suspend inline fun <T : Particle, U> T.withParticleContext(
+    crossinline block: suspend () -> U
+): U = withContext(handles.dispatcher) { block() }
```

Comment:
I'm conflicted about this.. on one level it will be easier to understand what's going on for people unfamiliar with coroutines, but on another - it's hiding what's going on. I wonder if it would be more appropriate to instead expose the dispatcher from the Particle. 

```kotlin
withContext(particle.dispatcher) {
    // ...
}
```

Additionally, I don't think we should be making it easier to run stuff outside the normal lifecycle of particles/handles. That type of access should largely be discouraged (where ingestion is really the only truly valid use-case).

Reply:
I don't think many people should be using this, outside of Arcs-Chronicle.

It provides us flexibility to change what "running in a particle context" actually means, implementation-wise.

Reply:
Maybe we can move it into a different package that's not part of the normal SDK import set, that arcs-chronicle can use?

Reply:
Would it be reasonable to use this if a particle wanted to dispatch some work off of the main scheduler context, and then when it was done, move back to the particle thread to store some results?

Reply:
That would be a good use-case... I think we should get the Handle Czar's input on this too. @mykmartin

Reply:
While that might be a good use case, I don't think the top-level particle api should expose this helper. I agree with Jason (no not that one the other one) that this is likely to encourage the wrong type of usage.

`I don't think many people should be using this, outside of Arcs-Chronicle.` -> That's definitely an indicator that this is the wrong place for the helper method.